src/os/bluestore/bluestore_types.cc

```diff
@@ -833,6 +838,15 @@ void bluestore_onode_t::generate_test_instances(list<bluestore_onode_t*>& o)
   // FIXME
 }
 
+size_t bluestore_onode_t::get_preferred_csum_order() const
+{
+  uint32_t t = expected_write_size;
+  if (!t) {
+    return 0;
+  }
+  return ctz(expected_write_size);
```

Comment:
yeah, right now the caller does

  wctx.csum_order = MAX(block_size_order, o->onode.get_preferred_csum_order());

because the onode_t doesn't know what the block_size is.